CascadeEnergy/aws-most
# SEN-1322 -- S3 listObjects as a monadic stream.

Had to write this one method when I was doing the downtime recovery script to pull files from s3 which had last modified times between a certain range.

I wanted this code in shareable spot, and I've also done some similar work with `sqs.receiveMessage` in our digest-chunker, so I thought I'd make `aws-most` it's own project. Also seems to mesh really well with my effort to promisificate all the aws clients in `aws-promised`.

**To Test**

Review code
Run the unit tests, verify coverage. `npm test`

Bonus points:

Make a testing bucket with some files, alter `examples/s3-listObjects.js` to make sense for the files in your bucket and run it.

test the pagination ability by putting more than a thousand files in a bucket and then running the listObjects stream.
